firefox-bootstrap/bootstrap.js

```diff
@@ -0,0 +1,224 @@
+    }
+  },
+
+  setWidgetState(options) {
+    let widget = CustomizableUI.getWidget(WIDGET_ID);
+    widget.disabled = options.disabled;
+  },
+
+  openWebCompatTab(args) {
+    const ACK_MESSAGE = "WebCompat:Ack";
+    const SCREENSHOT_MESSAGE = "WebCompat:SendScreenshot";
+    const FRAMESCRIPT = `chrome://webcompat-reporter/content/wc-frame.js?${Date.now()}`;
+    const WEBCOMPAT_ORIGIN = "https://webcompat.com";
+    let [browser, tabData] = args;
+
+    return new Promise((resolve) => {
+      let loadedListener = {
+        QueryInterface: XPCOMUtils.generateQI(["nsIWebProgressListener",
+          "nsISupportsWeakReference"]),
+        onStateChange: function(webProgress, request, flag, status) {
+          let ackFn;
+          if (flag & Ci.nsIWebProgressListener.STATE_STOP) {
+            if (request.originalURI.host === "webcompat.com") {
+              let mm = browser.selectedBrowser.messageManager;
+              browser.removeProgressListener(this);
+
+              // did we successfully get a screenshot?
+              if (tabData && tabData.canvasData) {
+                mm.loadFrameScript(FRAMESCRIPT, true);
+                mm.sendAsyncMessage(SCREENSHOT_MESSAGE, {
+                  screenshot: tabData.canvasData
+                });
+
+                mm.addMessageListener(ACK_MESSAGE, ackFn = (message) => {
+                  mm.removeMessageListener(ACK_MESSAGE, ackFn);
+                  resolve();
+                });
+              } else {
+                resolve();
+              }
+            }
+          }
+        }
+      };
+
+      browser.addProgressListener(loadedListener);
+      let encodedURL = encodeURIComponent(tabData.url);
+      browser.loadOneTab(`${WEBCOMPAT_ORIGIN}/issues/new?url=${encodedURL}`,
+        {inBackground: false});
+    });
+  },
+
+  getScreenshot(browser) {
+    const FRAMESCRIPT = `chrome://webcompat-reporter/content/tab-frame.js?${Date.now()}`;
+    const TABDATA_MESSAGE = "WebCompat:SendTabData";
+
+    return new Promise((resolve) => {
+      let receiveFn;
+      let mm = browser.selectedBrowser.messageManager;
+
+      mm.loadFrameScript(FRAMESCRIPT, true);
+      mm.addMessageListener(TABDATA_MESSAGE, receiveFn = (message) => {
+        mm.removeMessageListener(TABDATA_MESSAGE, receiveFn);
+        resolve([browser, message.json]);
+      });
+    });
+  },
+
+  reportIssue(xulDoc) {
+    Promise.resolve(xulDoc.defaultView.gBrowser).then(this.getScreenshot)
+                                                .then(this.openWebCompatTab)
+                                                .catch(Cu.reportError);
+  }
+};
+
+// use enabled pref as a way for tests (e.g. test_contextmenu.html) to disable
+// the addon when running.
+let prefObserver = function(aSubject, aTopic, aData) {
+  let enabled = Services.prefs.getBoolPref(PREF_WC_REPORTER_ENABLED);
+  if (enabled) {
+    WebCompatReporter.init();
+  } else {
+    WebCompatReporter.uninit();
+    WebCompatReporter = null;
```

Comment:
When disabling the preference and immediately re-enabling it, it would fail since you killed `WebCompatReporter` here.